### PR TITLE
Improvements to the ARM client factory

### PR DESCRIPTION
**Pull request by jhendrixMSFT**
Share the *arm.Client across SDK clients instead of always creating a new one (and not throwing away the "test" one).

### CODE REVIEW

**Comment by jhendrixMSFT** [comment]
@tadelesh I'll wait for your ack before merging. Also will clean up CODEOWNERS.

**Review by tadelesh — APPROVED** [review]

**Pull request merged**
